### DISCUSSION TITLE
How do I automatically move notes to the right folder every day?

### DISCUSSION THREAD

**Discussion by Pesssimist** [SilentVoid13/Templater#1258] [Help] [answered]
Hi!
I am trying to create a system in which each event will be recorded in a separate note. To do this, I create a note using a unique identifier, and use a templater. However, I want these notes to be automatically transferred when created to the folder of the corresponding day, in the following structure like: _everyday micro/12/06_december/note name.md_
For this I use the following template
```
- *[[<% tp.date.now("YYYY-MM-DD")%>]] <% tp.date.now("HH-mm")%>*
    - <%tp.file.cursor(1)%>
<% await tp.file.move("/everyday micro/2023/12/06_december/" + tp.file.title)%>
```

I want to write the date in the folder address (/everyday micro/2023/12/ 06_december/) using templater commands, so that a new folder with the date is created every day. But using the following command
`<% <% await tp.file.move("/everyday micro/<%tp.date.now("YYYY")%>/<% tp.date.now("MM")%>/<% tp.date.now("DD_MMMM")%>"+ tp.file.title)%>`
causes an error.
Please help!

**Comment by Zachatoo** [comment, marked as answer]
A good rule of thumb is to never nest command tags `<% %>`. If you're in a command, you can just call the Templater functions.

You can concatenate the pieces you want like this:
```js
<% await tp.file.move("/everyday micro/" + tp.date.now("YYYY") + "/" + tp.date.now("MM") + "/" + tp.date.now("DD_MMMM") + tp.file.title) %>
```

Or, using [template literals](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Template_literals):
```js
<% await tp.file.move(`/everyday micro/${tp.date.now("YYYY")}/${tp.date.now("MM")}/${tp.date.now("DD_MMMM")${tp.file.title}`) %>
```

**Reply by Pesssimist** [reply]
Thank you very much!
I tried options without `<% %>`, but since I don't understand the code, the most I could do was just a error when I tried to assemble the entire path to the file from `tp.date.now` blocks, like `("/everyday micro/tp.date.now("YYYY")/tp.date.now("MM"/tp.date.now("DD_MMMM")/ + tp.file.title)`

I used your solution
`<% await tp.file.move("/everyday micro/" + tp.date.now("YYYY") + "/" + tp.date.now("MM") + "/" + tp.date.now("DD_MMMM") + "/" + tp.file.title) %>`

Just added another one `+ "/"` before `+ tp.file.title`. This allows me to automatically create year/month/day subfolders in obsidian that contain files YYYYMMDD-HHmmss.md
